src/components/atoms/PasswordField.tsx

```diff
@@ -0,0 +1,78 @@
+import { ErrorMessage } from "@hookform/error-message";
+import { TextField } from "@mui/material";
+import { useState } from "react";
+import { FieldPath, RegisterOptions, useFormContext } from "react-hook-form";
+import InputAdornment from "@mui/material/InputAdornment";
+import Visibility from "@mui/icons-material/Visibility";
+import VisibilityOff from "@mui/icons-material/VisibilityOff";
+import IconButton from "@mui/material/IconButton";
+import { PasswordFormValues } from "../../types/FormTypes";
+import { useTranslation } from "react-i18next";
+
+interface PasswordFieldProps {
+  field: FieldPath<PasswordFormValues>;
+  label: string;
+  validationSchema?: RegisterOptions<
+    PasswordFormValues,
+    "password" | "confirmPassword"
+  >;
+}
+
+const PasswordField = ({
+  field,
+  label,
+  validationSchema,
+}: PasswordFieldProps) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const { t } = useTranslation();
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext<PasswordFormValues>();
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
+  const handleMouseDownPassword = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+  };
```

Comment:
Why do you need this? 
What happens on mouse down? 🤔

Reply:
onMouseDown allows the user to continue entering the password even when the password was shown/hidden by clicking the icon (thanks to the `event.preventDefault` in the handler). Without it, the field loses its focus, and the user has to manually click on the field again to start entering the password.

Reply:
Nice